xbmc/filesystem/FavouritesDirectory.cpp

```diff
@@ -190,8 +192,12 @@ std::string CFavouritesDirectory::GetExecutePath(const CFileItem &item, int cont
 std::string CFavouritesDirectory::GetExecutePath(const CFileItem &item, const std::string &contextWindow)
 {
   std::string execute;
-  if (item.m_bIsFolder && (g_advancedSettings.m_playlistAsFolders ||
-                            !(item.IsSmartPlayList() || item.IsPlayList())))
+  if (item.GetPath().substr(0, 13) == "favourites://")
+  {
+    execute = CURL::Decode(item.GetPath().substr(13));
+  }
```

Comment:
we have URIUtils::IsProtocol and ::GetFileName for this (no need for magic numbers)